### PR TITLE
Create solidity web app

### PR DESCRIPTION
**Pull request by yann300**
Here's its ipfs hash QmaHfk3y3pgUSKim9mBfqyae11uNm2qaigJsigzpqDvrK5

### CODE REVIEW

**Comment by shahnawazalam0996** [comment]
Ethereum and monero Web miner  and app miner for android 10 OS

**Comment by shahnawazalam0996** [comment]
Fake miner dond send me any mail

File: apps/remix-ide/src/app/tabs/styles/compile-tab-styles.js [hunk 1]
```diff
@@ -1,239 +1,11 @@
-const csjs = require('csjs-inject')
+var csjs = require('csjs-inject')
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/tabs/styles/compile-tab-styles.js:1]: Why `var`?

File: apps/solidity-compiler/src/app/app.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import { SolidityCompiler } from '@remix-ui/solidity-compiler' // eslint-disable-line
+
+import { CompilerClientApi } from './compiler'
+
+const remix = new CompilerClientApi()
+
+export const App = () => {  
+  return (
+    <div className="debugger">
```
Comment (Aniket-Engg) [inline apps/solidity-compiler/src/app/app.tsx:11]: should the className be the `debugger` here?

File: apps/solidity-compiler/src/app/compiler-api.ts [hunk 3]
```diff
@@ -0,0 +1,259 @@
+import { compile } from '@remix-project/remix-solidity'
+import { CompileTabLogic, parseContracts } from '@remix-ui/solidity-compiler' // eslint-disable-line
+import { ConfigurationSettings } from '@remix-project/remix-lib-ts'
```
Comment (Aniket-Engg) [inline apps/solidity-compiler/src/app/compiler-api.ts:3]: may be you can use `import type`

File: apps/solidity-compiler/src/app/compiler-api.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    this.compileErrors = {}
+    this.compiledFileName = ''
+    this.currentFile = ''
+  }
+
+  onActivation () {
+    this.listenToEvents()    
+  }
+
+  onDeactivation () {}
```
Comment (Aniket-Engg) [inline apps/solidity-compiler/src/app/compiler-api.ts:51]: This should have the logic to remove all listeners

File: apps/solidity-compiler/src/app/compiler-api.ts [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+  resolveContentAndSave (url) {
+    return this.call('contentImport', 'resolveAndSave', url)
```
Comment (Aniket-Engg) [inline apps/solidity-compiler/src/app/compiler-api.ts:54]: Check if `await` should be there

File: apps/solidity-compiler/src/app/compiler-api.ts [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+  }
+
+  compileWithHardhat (configFile) {
+    return this.call('hardhat', 'compile', configFile)
```
Comment (Aniket-Engg) [inline apps/solidity-compiler/src/app/compiler-api.ts:58]: Check if await should be there and same for others

File: apps/solidity-compiler/src/app/compiler-api.ts [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+   * set the compiler configuration
+   * This function is used by remix-plugin compiler API.
+   * @param {object} settings {evmVersion, optimize, runs, version, language}
+   */
+  setCompilerConfig (settings) {
+    this.configurationSettings = settings    
+  }  
+
+  getFileManagerMode () {
+    return 'browser'
```
Comment (Aniket-Engg) [inline apps/solidity-compiler/src/app/compiler-api.ts:124]: should this always return `browser`?

File: apps/solidity-compiler/src/app/compiler.ts [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  'hideWarnings': () => false,
+  'autoCompile': () => false,
+  'includeNightlies': () => false
+}
+
+const defaultCompilerParameters = {
+  runs: '200',
+  optimize: false,
+  version: 'soljson-v0.8.7+commit.e28d00a7',
+  evmVersion: null, // default
```
Comment (Aniket-Engg) [inline apps/solidity-compiler/src/app/compiler.ts:30]: may be it should be `compiler default`

File: apps/solidity-compiler/src/app/compiler-api.ts (hunk 5 above)
Comment (yann300) [reply apps/solidity-compiler/src/app/compiler-api.ts:54]: this returns a Promise,  `resolveContentAndSave` is declared with the following interface: `resolveContentAndSave: (url: string) => Promise<string>`

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**